packages/web-components/src/components/quote/quote.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { customElement, html, LitElement, property } from 'lit-element';
+import settings from 'carbon-components/es/globals/js/settings';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings';
+import styles from './quote.scss';
+import StableSelectorMixin from '../../globals/mixins/stable-selector';
+import '../horizontal-rule/horizontal-rule';
+
+const { prefix } = settings;
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+export enum QUOTE_TYPES {
+  /**
+   * Default - doubleCurved
+   */
+  DEFAULT = 'doubleCurved',
+
+  /**
+   * singleCurved
+   */
+  SINGLE_CURVED = 'singleCurved',
+
+  /**
+   * singleAngle
+   */
+  SINGLE_ANGLE = 'singleAngle',
+
+  /**
+   * doubleAngle
+   */
+  DOUBLE_ANGLE = 'doubleAngle',
+
+  /**
+   * lowHighReversedDoubleCurved
+   */
+  LOW_HIGH_REVERSED_DOUBLE_CURVED = 'lowHighReversedDoubleCurved',
+}
+
+const slotExistencePropertyNames = {
+  sourceHeading: '_hasSourceHeading',
+  sourceCopy: '_hasSourceCopy',
+  sourceBottomCopy: '_hasSourceBottomCopy',
+  cta: '_hasCTA',
+};
+
+/**
+ * Quote.
+ *
+ * @element dds-quote
+ */
+@customElement(`${ddsPrefix}-quote`)
+class DDSQuote extends StableSelectorMixin(LitElement) {
+  @property({ reflect: true, attribute: 'mark-type' })
+  markType = QUOTE_TYPES.DEFAULT;
+
+  @property({ reflect: true })
+  inverse: boolean = false;
+
+  /**
+   * `true` if there is source heading.
+   */
+  protected _hasSourceHeading = false;
+
+  /**
+   * `true` if there is source copy.
+   */
+  protected _hasSourceCopy = false;
+
+  /**
+   * `true` if there is source bottom copy.
+   */
+  protected _hasSourceBottomCopy = false;
+
+  /**
+   * `true` if there is cta.
+   */
+  protected _hasCTA = false;
+
+  /**
+   * Handles `slotchange` event.
+   */
+  protected _handleSlotChange({ target }: Event) {
+    const { name } = target as HTMLSlotElement;
+    const hasContent = (target as HTMLSlotElement)
+      .assignedNodes()
+      .some(node => node.nodeType !== Node.TEXT_NODE || node!.textContent!.trim());
+    this[slotExistencePropertyNames[name]] = hasContent;
+    this.requestUpdate();
+  }
+
+  updated() {
+    const { inverse } = this;
+    this.classList.toggle(`${prefix}--quote__inverse`, inverse);
+  }
```

Comment:
To make the DOM change happen only when `inverse` attribute/property changes:

```suggestion
  updated(changedProperties) {
    if (changedProperties.has('inverse')) {
      const { inverse } = this;
      this.classList.toggle(`${prefix}--quote__inverse`, inverse);
    }
  }
```